Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/QuickPulseProcessProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib;
+
+    /// <summary>
+    /// Top CPU collector.
+    /// </summary>
+    internal sealed class QuickPulseProcessProvider : IQuickPulseProcessProvider
+    {
+        private readonly IQuickPulsePerfLib perfLib;
+
+        public QuickPulseProcessProvider(IQuickPulsePerfLib perfLib)
+        {
+            this.perfLib = perfLib ?? PerfLib.PerfLib.GetPerfLib();
+        }
+
+        public void Initialize()
+        {
+            this.perfLib.Initialize();
+        }
+
+        public void Close()
+        {
+            this.perfLib.Close();
+        }
+
+        /// <summary>
+        /// Gets a collection of <see cref="QuickPulseProcess"/> objects - each corresponding to a system process and containing
+        /// information about the amount of time the process has occupied CPU cores.
+        /// </summary>
+        /// <param name="totalTime">If available, contains the value of the _Total instance of the counter, which indicates the overall
+        /// amount of time spent by CPU cores executing system processes.</param>
+        public IEnumerable<QuickPulseProcess> GetProcesses(out TimeSpan? totalTime)
+        {
+            const int CategoryIndex = 230;
+            const int CounterIndex = 6;
```

Comment:
Why 6, why 230?  Please doc comment your magic numbers so people will know what they mean in future.